contrib/brl/bseg/bapl/bapl_keypoint_extractor.cxx

```diff
@@ -373,13 +373,18 @@ bapl_lowe_orientation::orient_at( float x, float y, float scale,
   for (unsigned int i=0; i<peaks.size(); ++i) {
     if (histogram[ peaks[i] ] > max) {
       //parabolic interpolation
-      float ypos = histogram[ (peaks[i]+1)%num_bins_ ];
-      float yneg = histogram[ (peaks[i]-1)%num_bins_ ];
-      float dy   = (ypos - yneg)/2.0f;
-      float d2y  = 2.0f*histogram[ peaks[i] ] - ypos - yneg;
-      float dx = 6.28319f/num_bins_;
-      float angle = (float(peaks[i])+dy/d2y)*dx;
-      orientations.push_back(angle);
+      if(num_bins_ != 0) {
+        float ypos = histogram[ (peaks[i]+1)%num_bins_ ];
+        float yneg = histogram[ (peaks[i]-1)%num_bins_ ];
+        float dy   = (ypos - yneg)/2.0f;
+        float d2y  = 2.0f*histogram[ peaks[i] ] - ypos - yneg;
+        float dx = 6.28319f/num_bins_;
+        float angle = (float(peaks[i])+dy/d2y)*dx;
+        orientations.push_back(angle);
+      }
+      else {
+        vcl_cout << "ERROR: Division by 0" << vcl_endl;
```

Comment:
One thought--should execution stop entirely here?  It might be problematic if it continues while just printing out an error.  I don't know if these are used much in the library, but maybe this would constitute a "domain error."  https://github.com/vxl/vxl/blob/647386048f3f71ed6d9e7d67de5d11dab5ddacfc/vcl/generic/vcl_stdexcept.h